examples/wallet/mint_native_tokens.c

```diff
@@ -0,0 +1,181 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+/**
+ * @brief A simple example of minting a native tokens and sending them to a receiver address using wallet APIs.
+ *
+ */
+
+#include <inttypes.h>
+#include <stdio.h>
+#include <unistd.h>  // for Linux sleep()
+
+#include "wallet/output_alias.h"
+#include "wallet/output_foundry.h"
+#include "wallet/wallet.h"
+
+#define Mi 1000000
+
+#define NODE_HOST "localhost"
+#define NODE_PORT 14265
+#define NODE_USE_TLS false
+#define TEST_COIN_TYPE SLIP44_COIN_TYPE_IOTA
+
+// replace this with your mnemonic string
+static char const* const test_mnemonic =
+    "acoustic trophy damage hint search taste love bicycle foster cradle brown govern endless depend situate athlete "
+    "pudding blame question genius transfer van random vast";
+uint32_t const sender_addr_index = 0;      // address index of a sender
+uint32_t const state_ctrl_addr_index = 1;  // address index of a state controller
+uint32_t const govern_addr_index = 2;      // address index of a governor
+uint32_t const receiver_addr_index = 3;    // address index of a receiver of native tokens
+uint64_t const amount = 1;                 // transfer 1Mi from a sender to an alias output (address)
+static char const* const max_supply_str =
+    "1000000000000000000000000000000";                         // maximum supply of newly minted native tokens
+static char const* const minted_tokens_str = "1000000000000";  // number of newly minted native tokens
+
+int main(void) {
+  iota_wallet_t* w = wallet_create(test_mnemonic, "", TEST_COIN_TYPE, 0);
+  if (!w) {
+    printf("Failed to create a wallet object!\n");
+    return -1;
+  }
+
+  if (wallet_set_endpoint(w, NODE_HOST, NODE_PORT, NODE_USE_TLS) != 0) {
+    printf("Failed to set a wallet endpoint!\n");
+    wallet_destroy(w);
+    return -1;
+  }
+
+  if (wallet_update_node_config(w) != 0) {
+    printf("Failed to update a node configuration!\n");
+    wallet_destroy(w);
+    return -1;
+  }
+
+  address_t sender_addr, state_ctrl_addr, govern_addr, receiver_addr;
+  if (wallet_ed25519_address_from_index(w, false, sender_addr_index, &sender_addr) != 0) {
+    printf("Get sender address failed\n");
+    return -1;
+  }
+  if (wallet_ed25519_address_from_index(w, false, state_ctrl_addr_index, &state_ctrl_addr) != 0) {
+    printf("Get state controller address failed\n");
+    return -1;
+  }
+  if (wallet_ed25519_address_from_index(w, false, govern_addr_index, &govern_addr) != 0) {
+    printf("Get governor address failed!\n");
+    return -1;
+  }
+  if (wallet_ed25519_address_from_index(w, false, receiver_addr_index, &receiver_addr) != 0) {
+    printf("Get receiver address failed\n");
+    return -1;
```

Comment:
the wallet instance needs to be freed.

```suggestion
    printf("Get sender address failed\n");
    wallet_destroy(w);
    return -1;
  }
  if (wallet_ed25519_address_from_index(w, false, state_ctrl_addr_index, &state_ctrl_addr) != 0) {
    printf("Get state controller address failed\n");
    wallet_destroy(w);
    return -1;
  }
  if (wallet_ed25519_address_from_index(w, false, govern_addr_index, &govern_addr) != 0) {
    printf("Get governor address failed!\n");
    wallet_destroy(w);
    return -1;
  }
  if (wallet_ed25519_address_from_index(w, false, receiver_addr_index, &receiver_addr) != 0) {
    printf("Get receiver address failed\n");
    wallet_destroy(w);
    return -1;
```

Reply:
Thanks for catching this bug.
Done.